src/scripts/main.js

```diff
@@ -1,3 +1,207 @@
 'use strict';
 
 // write code here
+const headers = document.querySelectorAll('th');
+let rows = [...document.querySelectorAll('tbody tr')];
+const body = document.querySelector('tbody');
+const sortOrder = Array(headers.length).fill(true);
+
+function updateRows() {
+  rows = [...document.querySelectorAll('tbody tr')];
+}
+
+function sort(index) {
+  updateRows();
+
+  rows.sort((a, b) => {
+    const cellA = a.cells[index].textContent;
+    const cellB = b.cells[index].textContent;
+
+    if (cellA.includes('$')) {
+      const aValue = parseFloat(cellA.replace(/[$,]/g, ''));
+      const bValue = parseFloat(cellB.replace(/[$,]/g, ''));
+
+      return sortOrder[index] ? aValue - bValue : bValue - aValue;
+    } else {
+      return sortOrder[index]
+        ? cellA.localeCompare(cellB)
+        : cellB.localeCompare(cellA);
+    }
+  });
+
+  body.innerHTML = '';
+
+  rows.forEach((row) => body.append(row));
+
+  sortOrder[index] = !sortOrder[index];
+}
+
+headers.forEach((head, index) => {
+  head.addEventListener('click', () => {
+    sort(index);
+  });
+});
+
+rows.forEach((row) => {
```

Comment:
Instead of adding a listener for each row add it to the table body and use the event bubbling concept to do something with the row. [Event delegation](https://www.freecodecamp.org/news/event-delegation-javascript/)

Would be nice to do similar with headers